src/osd/OSD.cc

```diff
@@ -7621,10 +7621,9 @@ void OSD::resched_all_scrubs()
     if (!pg)
       continue;
 
-    if (!pg->get_planned_scrub().must_scrub && !pg->get_planned_scrub().need_auto) {
```

Comment:
I can't find that check (i.e. - making sure we are not rescheduling a priority scrub for resched_all_scrubs())
anywhere in the new code.

Reply:
We called reschedule_scrub unconditionally in update_stats -- if this check is actually necessary than it doesn't appear to have been sufficient anyway.  I'd expect update_scrub_job to preserve any priority without need for additional guards.

Reply:
Seems like update_scrub_job's behavior should be fine as is if these flags are indeed set?

Reply:
In the existing code (pre my suggested scheduling changes), there are some subtle changes that will result from trying to reschedule high-priority jobs. I will verify how the new code handles that.